packages/core/src/App/Containers/RealAccountSignup/account-wizard.jsx

```diff
@@ -70,47 +70,53 @@ class AccountWizard extends React.Component {
 
     componentDidMount() {
         this.props.fetchStatesList();
-        const { cancel, promise } = makeCancellablePromise(this.props.fetchResidenceList());
-        this.cancel = cancel;
-        const { cancel: cancelFinancialAssessment, promise: financial_assessment_promise } = makeCancellablePromise(
-            this.props.fetchFinancialAssessment()
+
+        const { cancel, promise } = makeCancellablePromise(
+            Promise.all([this.props.fetchResidenceList(), this.props.fetchFinancialAssessment()])
         );
-        this.cancelFinancialAssessment = cancelFinancialAssessment;
+        this.cancel = cancel;
```

Comment:
IIRC FA is required for some EU users, do we get that information in some other place?

Reply:
@reza-fs we get FA here. Combined both `fetchResidenceList` and `fetchFinancialAssessment` promises to make a `CancellablePromise`

Reply:
Ah, this is clever. Sorry I did not see it.